sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSPluginInternal.scala

```diff
@@ -210,7 +216,31 @@ private[sbtplugin] object ScalaJSPluginInternal {
             }
 
             Set(output, sourceMapFile)
-          } (realFiles.toSet)
+          }
+
+          def useMainChanged = (scalaJSUseMainModuleInitializerTask in key)
+            .previous
+            .exists(_ != (scalaJSUseMainModuleInitializerTask in key).value)
+
+          def moduleInitializersChanged = (scalaJSModuleInitializersFingerprints in key)
+            .previous
+            .exists(_ != (scalaJSModuleInitializersFingerprints in key).value)
+
+          def linkerConfigChanged = (scalaJSLinkerConfigFingerprint in key)
+            .previous
+            .exists(_ != (scalaJSLinkerConfigFingerprint in key).value)
+
+          val configChanged =
+            useMainChanged || moduleInitializersChanged || linkerConfigChanged
+
+          if (configChanged) {
+            link()
```

Comment:
Will the next run that uses `FileFunction.cached` work properly if you do not always use it?

Reply:
Hum, that is potentially an issue. I'm not sure how `FileFunction` is supposed to be used to get around that.

Perhaps one way to workaround this difficulty is to always use `FileFunction.cached()`, but before that do
```scala
if (configChanged && output.exists())
  output.delete()
```
?

Reply:
Ah, I see the problem. Currently, if I change a config setting *and* change some code, run `reload`, `fastOptJS` and `fastOptJS` again, it links twice: once because of the config change, and once because the cached file has changed. The compilation output stays correct, but we don't need to link the second time around, and can save some effort there.

I'll experiment with this and report back.

Reply:
The solution you suggested works fine, and is what I've implemented in 818ce79a384408cd7cd0ec51481b4966e26957ff